Crash Report: IllegalArgumentException FileUtils.contentQuery (FileUtils.java:159)

**Describe the bug**
```
java.lang.RuntimeException: 

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2756)

 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2825)

 
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)

 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1557)

 
  at android.os.Handler.dispatchMessage (Handler.java:110)

 
  at android.os.Looper.loop (Looper.java:203)

 
  at android.app.ActivityThread.main (ActivityThread.java:6339)

 
  at java.lang.reflect.Method.invoke (Native Method)

 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1084)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:945)
Caused by: java.lang.IllegalArgumentException: 

 
  at android.database.AbstractCursor.getColumnIndexOrThrow (AbstractCursor.java:333)

 
  at android.database.CursorWrapper.getColumnIndexOrThrow (CursorWrapper.java:87)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:159)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:144)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.manageExternalLaunchAndRestoringViewState (KiwixMobileActivity.java:1771)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate (KiwixMobileActivity.java:462)

 
  at android.app.Activity.performCreate (Activity.java:6689)

 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1140)

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2709)
```

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,10 @@
+2.5.2
+BUGFIX: Some zim files were not displaying
+
+2.5.1
+NEW: Downloads are now using the DownloadManager
+NEW: Downloads/Device/Library completely rewritten
+
 2.5
 NEW: Downloads are now using the DownloadManager
 NEW: Downloads/Device/Library completely rewritten

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,6 @@
+2.5.3
+BUGFIX: Fix IndexOutOfBounds in KiwixMobileActivity `onResume`
+
 2.5.2
 BUGFIX: Some zim files were not displaying
 

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -188,7 +188,7 @@ def buildNumber = System.getenv('TRAVIS_BUILD_NUMBER') ?: "dev"
 ext {
   versionMajor = 2
   versionMinor = 5
-  versionPatch = 2
+  versionPatch = 3
 }
 
 private String generateVersionName() {

--- a/app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java
@@ -1260,7 +1260,9 @@ public class KiwixMobileActivity extends BaseActivity implements WebViewCallback
       }
     }
 
-    if (!mWebViews.isEmpty() && mWebViews.get(currentWebViewIndex).getUrl() != null &&
+    if (!mWebViews.isEmpty() &&
+        currentWebViewIndex < mWebViews.size() &&
+        mWebViews.get(currentWebViewIndex).getUrl() != null &&
         mWebViews.get(currentWebViewIndex).getUrl().equals("file:///android_asset/help.html") &&
         mWebViews.get(currentWebViewIndex).findViewById(R.id.get_content_card) != null) {
       mWebViews.get(currentWebViewIndex).findViewById(R.id.get_content_card).setEnabled(true);

--- a/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java
@@ -36,7 +36,7 @@ import static org.kiwix.kiwixmobile.downloader.ChunkUtils.deleteAllParts;
 import static org.kiwix.kiwixmobile.utils.Constants.TAG_KIWIX;
 
 public class FileUtils {
-  private static final String TAG = "FileUtils";
+  private static final String TAG = "KiwixFileUtils";
 
   public static File getFileCacheDir(Context context) {
     boolean external = Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState());

--- a/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java
@@ -25,22 +25,18 @@ import android.os.Build;
 import android.os.Environment;
 import android.provider.DocumentsContract;
 import android.util.Log;
-
-import org.kiwix.kiwixmobile.BuildConfig;
-import org.kiwix.kiwixmobile.downloader.ChunkUtils;
-import org.kiwix.kiwixmobile.library.entity.LibraryNetworkEntity.Book;
-
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
+import org.kiwix.kiwixmobile.BuildConfig;
 
 import static org.kiwix.kiwixmobile.downloader.ChunkUtils.deleteAllParts;
 import static org.kiwix.kiwixmobile.utils.Constants.TAG_KIWIX;
 
 public class FileUtils {
+  private static final String TAG = "FileUtils";
 
   public static File getFileCacheDir(Context context) {
     boolean external = Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState());
@@ -150,17 +146,14 @@ public class FileUtils {
   }
 
   static private String contentQuery(Context context, Uri uri) {
-    Cursor cursor = null;
+    try (Cursor cursor = context.getContentResolver()
+        .query(uri, new String[] { "_data" }, null, null, null)) {
 
-    try {
-      cursor = context.getContentResolver().query(uri, new String[]{"_data"}, null, null, null);
-
-      if (cursor != null && cursor.moveToFirst())
+      if (cursor != null && cursor.moveToFirst()) {
         return cursor.getString(cursor.getColumnIndexOrThrow("_data"));
-
-    } finally {
-      if (cursor != null)
-        cursor.close();
+      }
+    } catch (IllegalArgumentException e) {
+      Log.e(TAG, "Uri: " + uri.toString(), e);
     }
 
     return null;

